### PR TITLE
A few storage-related tweaks

### PR DESCRIPTION
**Pull request by ljedrz**
This PR deduplicates some code and applies a few cleanups and performance tweaks in `snarkos-storage`. Each commit is a single self-contained change.

### CODE REVIEW

File: storage/src/state/ledger.rs [hunk 1]
```diff
@@ -231,11 +237,6 @@ impl<N: Network> LedgerState<N> {
         ledger.regenerate_ledger_tree()?;
         assert_eq!(ledger.ledger_tree.read().root(), latest_ledger_root);
 
-        // let value = storage.export()?;
-        // println!("{}", value);
-        // let storage_2 = S::open(".ledger_2", context)?;
-        // storage_2.import(value)?;
-
```
Comment (ljedrz) [inline storage/src/state/ledger.rs:238]: We are already testing this in https://github.com/AleoHQ/snarkOS/blob/testnet3/storage/src/storage/rocksdb/tests.rs#L212-L235.

**Review by niklaslong — APPROVED** [review]
LGTM 👍

File: storage/src/state/ledger.rs [hunk 2]
```diff
@@ -100,6 +100,20 @@ impl<N: Network> LedgerState<N> {
         Self::open_writer_with_increment::<S, P>(path, 10_000)
     }
 
+    fn new<S: Storage>(storage: S, is_read_only: bool) -> Result<LedgerState<N>> {
```
Comment (howardwu) [inline storage/src/state/ledger.rs]: Can we leave this undeduped?

Comment (ljedrz) [reply storage/src/state/ledger.rs]: Sure; let me know if the rest is good to go, and I can merge this myself after dropping this one commit.

Comment (howardwu) [reply storage/src/state/ledger.rs]: Yes good to go

**Pull request merged**